packages/nextjs/src/config/loaders/dataFetchersLoader.ts

```diff
@@ -94,44 +105,73 @@ function wrapFunctions(userCode: string, templateCode: string, filepath: string)
  * Wrap `getStaticPaths`, `getStaticProps`, and `getServerSideProps` (if they exist) in the given page code
  */
 export default function wrapDataFetchersLoader(this: LoaderThis<LoaderOptions>, userCode: string): string {
-  // We know one or the other will be defined, depending on the version of webpack being used
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  const { projectDir } = this.getOptions ? this.getOptions() : this.query!;
-
   // For now this loader only works for ESM code
   if (!isESM(userCode)) {
     return userCode;
   }
 
-  // If none of the functions we want to wrap appears in the page's code, there's nothing to do. (Note: We do this as a
-  // simple substring match (rather than waiting until we've parsed the code) because it's meant to be an
-  // as-fast-as-possible fail-fast. It's possible for user code to pass this check, even if it contains none of the
-  // functions in question, just by virtue of the correct string having been found, be it in a comment, as part of a
-  // longer variable name, etc. That said, when we actually do the code manipulation we'll be working on the code's AST,
-  // meaning we'll be able to differentiate between code we actually want to change and any false positives which might
-  // come up here.)
-  if (Object.keys(DATA_FETCHING_FUNCTIONS).every(functionName => !userCode.includes(functionName))) {
-    return userCode;
-  }
+  // We know one or the other will be defined, depending on the version of webpack being used
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { projectDir } = 'getOptions' in this ? this.getOptions() : this.query;
 
-  const templatePath = path.resolve(__dirname, '../templates/dataFetchersLoaderTemplate.js');
-  // make sure the template is included when runing `webpack watch`
-  this.addDependency(templatePath);
+  // Proxy the processed file
+  if (!this.resourceQuery.includes('sentry-proxy-loader')) {
+    const ast = makeAST(userCode, true); // is there a reason to ever parse without typescript?
```

Comment:
> // is there a reason to ever parse without typescript?

Have you tried parsing a JS file with the `tsx` parser? Does it work? If so, then no, probably not.

Reply:
The "files" (strings) we're parsing in the `ast.test.ts` tests are JS and JS is a subset of TS so I am pretty positive it would work. Should we remove the differentiation and just always parse with typescript? Wdyt?

Reply:
Discussed: Let's try a big JS file or two and see if it errs out. If it works lets simplify the makeAst logic.

Reply:
Done in https://github.com/getsentry/sentry-javascript/pull/5563.